docker/workspace-build/Dockerfile

```diff
@@ -108,6 +112,17 @@ COPY docker/cyclonedds_profile.xml.tmpl \
      docker/workspace_setenv.sh \
      "${ANGEL_WORKSPACE_DIR}"/
 
+# Build and install rosbag2 from foxy-future branch,
+# which includes rosbag2_py package
+# Unfortunately, ros-foxy-rosbag2 does not include rosbag2_py
+ARG ROSBAG_SRC_DIR
+ENV ROSBAG_SRC_DIR="${ROSBAG_SRC_DIR}"
+RUN source "/opt/ros/${ROS_DISTRO}/setup.bash" \
+    && mkdir "${ROSBAG_SRC_DIR}" \
+    && cd "${ROSBAG_SRC_DIR}" \
+    && git clone -b foxy-future https://github.com/ros2/rosbag2.git \
```

Comment:
Do we need this just for the python package (`rosbag2_py`)? We could just pull/build that one package if the others are satisfied by the system installs.

Reply:
Yeah, just for `rosbag2_py`. What do you mean by only pulling that one package? I think the only way for us to get `rosbag2_py` with the foxy distro is by using this foxy-future branch of the rosbag2 repo and building from source.

Reply:
I was maybe more confident in my phrasing than I meant to be. I more intended to ask if it would make sense to just pull the rosbag2_py component to build in our workspace. It seems to only depend on packages that are currently installed in the container. I wasn't sure if there are major version differences between 0.3.8 --> 0.7.0. Upon reading the modified README and the associated ROS discourse post, I understand that we would need everything due to ABI/API changes to the I assume the surrounding/dependent packages.

I still think pulling these packages in as a submodule of the repo is more appropriate than cloning/building in the container build phase.